src/java.base/share/classes/jdk/internal/util/random/RandomSupport.java

```diff
@@ -1186,10 +1186,10 @@ public static double computeNextExponential(RandomGenerator rng) {
                 // For the exponential distribution, every overhang is convex.
                 final double[] X = DoubleZigguratTables.exponentialX;
                 final double[] Y = DoubleZigguratTables.exponentialY;
-                for (;; U1 = (rng.nextLong() >>> 1)) {
+                // At this point, the high-order bits of U1 have not been used yet,
+                // but we need the value in U1 to be positive.
+                for (U1 = (U1 >>> 1);; U1 = (rng.nextLong() >>> 1)) {
```

Comment:
A minor thing, but I would probably write `U1 >>>= 1` instead of `U1 = (U1 >>> 1)`.